bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,326 @@
+* User application (WAR, JAR, EAR) unless a hollow bootable JAR is created.
+* Runtime to launch server.
+
+The plugin outputs a JAR named <project>-wildfly.jar
+
+===== Bootable JAR runtime resolution
+
+The set of classes that implement the boot logic are retrieved at packaging time. 
+The new “org.wildfly.core:wildfly-jar-boot” JAR artifact (implementation located in wildfly-core repository) 
+is resolved thanks to the wildfly-core version of the provisioned server and shaded into the bootable JAR. 
+If the artifact doesn’t exist, the packaging will abort, the server being packaged not supporting bootable 
+JAR packaging. (e.g.: old release not containing new artifacts).
+
+This artifact is in charge to unzip the server and setup a JBOSS_HOME. Once this is done, 
+it setups JBoss module classloader and call into a new JBoss module “org.wildfly.bootable-jar” 
+that is in charge of the actual server run.  This new JBoss module implementation located in wildfly-core, 
+(“org.wildfly.core:wildfly-jar-runtime” artifact), is expected to be automatically provisioned by Galleon in all cases. 
+If the module is not part of the provisioned server (for some un-expected reasons), the server execution aborts.
+
+===== Galleon provisioning
+
+Galleon provisions a server thanks to the WildFly Galleon feature-pack.
+Galleon provisioning is operated inside the Maven plugin. There are 2 configuration ways:
+
+* Galleon configuration as Maven plugin configuration items:
+** A Galleon feature-pack location that identifies the server and version. This is required if no provisioning.xml file is provided nor feature-packs list.
+** A list of feature-packs (if no feature-pack location has been set).
+** Optionally, a list of layers to include.
+** Optionally, a list of layers to exclude.
+* Galleon provisioning.xml file located in <app src>/galleon/provisioning.xml. NB: Maven plugin configuration items (if set) override provisioning.xml file. 
+Using provisioning.xml file, custom feature-packs can be combined during the provisioning phase to provision modules, features, content (e.g.: DB drivers).
+
+If no Galleon layers are specified, then a configuration identical to the default _standalone-microprofile.xml_ is provisioned.
+
+===== CLI script execution
+
+When building the bootable JAR, a path to a set of CLI scripts can be provided to update the server configuration packaged in the bootable JAR. The CLI scripts are concatenated and executed once the server is provisioned and deployment copied into it (if any).
+Having multiple CLI scripts provides enough flexibility to reuse scripts in different contexts.
+The plugin can be configured with a CLI properties file to resolve properties present in CLI scripts. 
+This is similar to the _--properties=<properties file>_ option that you can provide to JBoss CLI command line. 
+This http://www.mastertheboss.com/jboss-server/jboss-script/using-properties-in-cli-scripts[article] covers the feature.
+
```

Comment:
Again here I think it would be good to differentiate between executing CLI scripts at runtime vs executing CLI scripts during server packaging.
I think athat even just a line with a link pointing to the community doc would work.

Reply:
Ok.